src/settings/target/mod.rs

```diff
@@ -140,27 +156,48 @@ impl Manifest {
 
     pub fn generate(
         name: String,
-        target_type: TargetType,
+        target_type: Option<TargetType>,
         config_path: PathBuf,
         site: Option<Site>,
     ) -> Result<Manifest, failure::Error> {
+        let config_file = config_path.join("wrangler.toml");
+        let template_config_content = fs::read_to_string(&config_file);
+        let template_config = match &template_config_content {
+            Ok(content) => {
+                let config: TemplateConfig = toml::from_str(content)?;
+                if let Some(target_type) = target_type.clone() {
+                    if config.target_type != target_type {
+                        message::warn(&format!("The template recommends the \"{}\" type. Using type \"{}\" may cause errors, we recommend changing the type field in wrangler.toml to \"{}\"", config.target_type, target_type, config.target_type));
+                    }
+                }
+                Ok(config)
+            }
+            Err(err) => Err(err),
+        };
+        let template_config = match template_config {
+            Ok(config) => config,
+            Err(err) => {
```

Comment:
I feel like we shouldn't punish the user for something they likely didn't configure - the error should be in the debug logs but otherwise it will fall back to the default which is `webpack_config`. This gives wrangler some flexibility when it comes to generating from templates that may not be maintained by us and don't contain a `wrangler.toml`

Reply:
I feel like being explicit about the error would help users know when a template is broken and requires a ticket/issue... what do you think?

Reply:
Iiiii think I disagree. Don't think we should block our users for something that's probably not their fault and I also don't think we should require template creators to have a `wrangler.toml` with these values filled in.